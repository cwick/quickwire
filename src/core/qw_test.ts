import {
  describe,
  it,
  mockPlatform,
  expect,
  assertSnapshot,
} from "./testing.ts";
import QW from "./qw.ts";

describe("QW", () => {
  it("prints the version number and exits", () => {
    const platform = mockPlatform();
    new QW(platform).run("version");
    expect(platform.log).toHaveBeenCalledWith("qw (quickwire) test_version");
    expect(platform.exit).toHaveBeenCalledWith(0);
  });

  it("prints usage information and exits", async (t) => {
    const platform = mockPlatform();
    let stdout = "";
    platform.log = (message: string) => (stdout += message + "\n");

    new QW(platform).run("help");
    await assertSnapshot(t, stdout);
    expect(platform.exit).toHaveBeenCalledWith(0);
  });

  it("exports a project", async () => {
    const platform = mockPlatform();
    platform.import = (path: string) => {
      if (path === "routes/index.tsx") {
        return Promise.resolve({ default: () => "Hello world" });
      }
      throw new Error(`Unexpected import: ${path}`);
    };

    await new QW(platform).run("export");
    expect(platform.writeTextFile).toHaveBeenCalledWith(
      "index.html",
      "Hello world"
    );
    expect(platform.exit).toHaveBeenCalledWith(0);
  });
});
