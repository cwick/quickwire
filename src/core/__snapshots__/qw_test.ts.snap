export const snapshot = {};

snapshot[`QW > prints usage information and exits 1`] = `
"Usage: qw [options] [COMMAND]

Commands:
  init	Create a new Quickwire project (not implemented)
  export	Export the current project
  start	Start the current project

Options:
  --help     Show this message and exit.
  --version  Print version information and exit.
"
`;
